File: finalize_docs.py
import base64
import os
import subprocess
import json

files = [
    "README.md",
    "docs/cloudflare_overview.md",
    "docs/development_workflow.md",
    "docs/setup_deployment.md",
    "docs/environments.md",
    "docs/images/qraft_logo.png",
    "docs/images/qraft_infrastructure_concept.png",
    "server/package.json",
    "server/tsconfig.json"
]

branch = "final-docs-v8"
repo = "junichi-muraoka/ut-qms"

def run_gh(args):
    result = subprocess.run(["gh"] + args, capture_output=True, text=True)
    return result

# 1. Get main SHA
res = run_gh(["api", f"repos/{repo}/git/refs/heads/main", "--jq", ".object.sha"])
main_sha = res.stdout.strip()
print(f"Main SHA: {main_sha}")

# 2. Create branch
run_gh(["api", f"repos/{repo}/git/refs", "--method", "POST", "-f", f"ref=refs/heads/{branch}", "-f", f"sha={main_sha}"])
print(f"Created branch {branch}")

# 3. Upload files
for f in files:
    if os.path.exists(f):
        with open(f, "rb") as file:
            content = base64.b64encode(file.read()).decode("utf-8")
        
        # Get current SHA if exists
        res = run_gh(["api", f"repos/{repo}/contents/{f}?ref={branch}", "--jq", ".sha"])
        sha = res.stdout.strip()
        
        args = ["api", "--method", "PUT", f"/repos/{repo}/contents/{f}", "-f", f"message=docs: finalize {f}", "-f", f"content={content}", "-f", f"branch={branch}"]
        if sha:
            args += ["-f", f"sha={sha}"]
        
        res = run_gh(args)
        if res.returncode == 0:
            print(f"Successfully uploaded {f}")
        else:
            print(f"Failed to upload {f}: {res.stderr}")

# 4. Create PR
run_gh(["pr", "create", "--title", "Final Documentation Overhaul (V8)", "--body", "Complete professional update", "--base", "main", "--head", branch])
print("Created PR")

# 5. Merge PR
run_gh(["pr", "merge", "--merge", "--admin", "--delete-branch=false"])
print("Attempted Merge")

File: finalize_docs.js
const fs = require('fs');
const { execSync } = require('child_process');

const files = [
    "README.md",
    "docs/cloudflare_overview.md",
    "docs/development_workflow.md",
    "docs/setup_deployment.md",
    "docs/environments.md",
    "docs/images/qraft_logo.png",
    "docs/images/qraft_infrastructure_concept.png",
    "server/package.json",
    "server/tsconfig.json"
];

const branch = "final-docs-v9";
const repo = "junichi-muraoka/ut-qms";

function runGh(args) {
    try {
        return execSync(`gh ${args.join(' ')}`).toString().trim();
    } catch (e) {
        return null;
    }
}

// 1. Get main SHA
const mainSha = runGh(["api", `repos/${repo}/git/refs/heads/main`, "--jq", ".object.sha"]);
console.log(`Main SHA: ${mainSha}`);

// 2. Create branch
runGh(["api", `repos/${repo}/git/refs`, "--method", "POST", "-f", `ref=refs/heads/${branch}`, "-f", `sha=${mainSha}`]);
console.log(`Created branch ${branch}`);

// 3. Upload files
for (const f of files) {
    if (fs.existsSync(f)) {
        const content = fs.readFileSync(f).toString('base64');
        
        // Get current SHA if exists
        const sha = runGh(["api", `repos/${repo}/contents/${f}?ref=${branch}`, "--jq", ".sha"]);
        
        let args = ["api", "--method", "PUT", `/repos/${repo}/contents/${f}`, "-f", `message="docs: finalize ${f}"`, "-f", `content="${content}"`, "-f", `branch="${branch}"`];
        if (sha) {
            args.push("-f", `sha="${sha}"`);
        }
        
        try {
            execSync(`gh ${args.join(' ')}`);
            console.log(`Successfully uploaded ${f}`);
        } catch (e) {
            console.log(`Failed to upload ${f}`);
        }
    }
}

// 4. Create PR
runGh(["pr", "create", "--title", '"Final Documentation Overhaul (V9)"', "--body", '"Complete professional update"', "--base", "main", "--head", branch]);
console.log("Created PR");

// 5. Merge PR
runGh(["pr", "merge", "--merge", "--admin", "--delete-branch=false"]);
console.log("Attempted Merge");
